Author: lluis
Date: 2009-10-21 12:21:58 -0400 (Wed, 21 Oct 2009)
New Revision: 144546

Modified:
   trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/ChangeLog
   trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/MonoDevelop.GtkCore.GuiBuilder/GuiBuilderProject.cs
   trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/MonoDevelop.GtkCore.csproj
   trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/gtk-gui/MonoDevelop.GtkCore.Dialogs.GtkDesignerOptionsPanelWidget.cs
   trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/gtk-gui/generated.cs
   trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/libstetic/ChangeLog
   trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/libstetic/wrapper/Container.cs
   trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/libstetic/wrapper/Widget.cs
   trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/libsteticui/ChangeLog
   trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/libsteticui/libsteticui.csproj
Log:
* gtk-gui/generated.cs:
* MonoDevelop.GtkCore.csproj:
* libsteticui/libsteticui.csproj:
* gtk-gui/MonoDevelop.GtkCore.Dialogs.GtkDesignerOptionsPanelWidget.cs:
  Flush.

* libstetic/wrapper/Widget.cs: Don't fire the name changed event while
  loading.

* libstetic/wrapper/Container.cs: Set the loading flag when loading.

* MonoDevelop.GtkCore.GuiBuilder/GuiBuilderProject.cs: Use typed
  collection instead of ArrayList.

Modified: trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/MonoDevelop.GtkCore.GuiBuilder/GuiBuilderProject.cs
===================================================================
Author: lluis
Date: 2009-10-21 12:21:58 -0400 (Wed, 21 Oct 2009)
New Revision: 144546

Modified:
   trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/ChangeLog
   trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/MonoDevelop.GtkCore.GuiBuilder/GuiBuilderProject.cs
   trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/MonoDevelop.GtkCore.csproj
   trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/gtk-gui/MonoDevelop.GtkCore.Dialogs.GtkDesignerOptionsPanelWidget.cs
   trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/gtk-gui/generated.cs
   trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/libstetic/ChangeLog
   trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/libstetic/wrapper/Container.cs
   trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/libstetic/wrapper/Widget.cs
   trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/libsteticui/ChangeLog
   trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/libsteticui/libsteticui.csproj
Log:
* gtk-gui/generated.cs:
* MonoDevelop.GtkCore.csproj:
* libsteticui/libsteticui.csproj:
* gtk-gui/MonoDevelop.GtkCore.Dialogs.GtkDesignerOptionsPanelWidget.cs:
  Flush.

* libstetic/wrapper/Widget.cs: Don't fire the name changed event while
  loading.

* libstetic/wrapper/Container.cs: Set the loading flag when loading.

* MonoDevelop.GtkCore.GuiBuilder/GuiBuilderProject.cs: Use typed
  collection instead of ArrayList.

Modified: trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/MonoDevelop.GtkCore.GuiBuilder/GuiBuilderProject.cs
===================================================================
--- trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/MonoDevelop.GtkCore.GuiBuilder/GuiBuilderProject.cs	2009-10-21 14:26:30 UTC (rev 144545)
+++ trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/MonoDevelop.GtkCore.GuiBuilder/GuiBuilderProject.cs	2009-10-21 16:21:58 UTC (rev 144546)
@@ -52,7 +52,7 @@
 	{
 		internal object MemoryProbe = Counters.GuiProjectsInMemory.CreateMemoryProbe ();
 		
-		ArrayList formInfos;
+		List<GuiBuilderWindow> formInfos;
 		Stetic.Project gproject;
 		DotNetProject project;
 		string fileName;
@@ -84,7 +84,7 @@
 				return;
 			
 			gproject = GuiBuilderService.SteticApp.CreateProject ();
-			formInfos = new ArrayList ();
+			formInfos = new List<GuiBuilderWindow> ();
 			
 			if (!System.IO.File.Exists (fileName)) {
 				// Regenerate the gtk-gui folder if the stetic project
@@ -223,7 +223,7 @@
 			}
 		}
 		
-		public ICollection Windows {
+		public ICollection<GuiBuilderWindow> Windows {
 			get {
 				Load ();
 				return formInfos; 

Modified: trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/libstetic/wrapper/Container.cs
===================================================================
--- trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/libstetic/wrapper/Container.cs	2009-10-21 14:26:30 UTC (rev 144545)
+++ trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/libstetic/wrapper/Container.cs	2009-10-21 16:21:58 UTC (rev 144546)
@@ -302,10 +302,15 @@
 				}
 			}
 			
-			ReadActionGroups (reader, elem);
-			ReadProperties (reader, elem);
-			ReadChildren (reader, elem);
-			DoSync ();
+			try {
+				Loading = true;
+				ReadActionGroups (reader, elem);
+				ReadProperties (reader, elem);
+				ReadChildren (reader, elem);
+				DoSync ();
+			} finally {
+				Loading = false;
+			}
 		}
 		
 		protected virtual void ReadChildren (ObjectReader reader, XmlElement elem)

Modified: trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/libstetic/wrapper/Widget.cs
===================================================================
--- trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/libstetic/wrapper/Widget.cs	2009-10-21 14:26:30 UTC (rev 144545)
+++ trunk/monodevelop/main/src/addins/MonoDevelop.GtkCore/libstetic/wrapper/Widget.cs	2009-10-21 16:21:58 UTC (rev 144546)
@@ -883,7 +883,8 @@
 						
 					string on = oldName;
 					oldName = Wrapped.Name;
-					OnNameChanged (new WidgetNameChangedArgs (this, on, Wrapped.Name));
+					if (!Loading)
+						OnNameChanged (new WidgetNameChangedArgs (this, on, Wrapped.Name));
 					
 					// Keep the member name in sync with the widget name
 					if (on == MemberName)
@@ -895,7 +896,8 @@
 					base.EmitNotify (propertyName);
 					string on = oldMemberName;
 					oldMemberName = MemberName;
-					OnMemberNameChanged (new WidgetNameChangedArgs (this, on, MemberName));
+					if (!Loading)
+						OnMemberNameChanged (new WidgetNameChangedArgs (this, on, MemberName));
 				}
 			}
 			else {
